plugins/techdocs-backend/src/techdocs/stages/generate/types.ts

```diff
@@ -26,14 +27,18 @@ export type GeneratorRunResult = {
 };
 
 /**
- * The values that the generator will receive. The directory of the
- * uncompiled documentation, with the values from the frontend. A dedicated log stream and a docker
- * client to run any generator on top of your directory.
+ * The values that the generator will receive.
+ *
+ * @param {string} directory The directory of the uncompiled documentation, with the values from the frontend
+ * @param {Docker} dockerClient A docker client to run any generator on top of your directory
+ * @param {ParsedLocationAnnotation} parsedLocationAnnotation backstage.io/techdocs-ref annotation of an entity
+ * @param {Writable} [logStream] A dedicated log stream
  */
 export type GeneratorRunOptions = {
   directory: string;
-  logStream?: Writable;
   dockerClient: Docker;
+  parsedLocationAnnotation: ParsedLocationAnnotation;
```

Comment:
Might make sense to pass in the entire entity here. It's not necessary to change it now i think, but might want to do some other fancy sorcery later on.

Reply:
Yeah agreed!